### PR TITLE
caffe: init at git-rev 77d66dfc

### PR DESCRIPTION
**Pull request by jb55**
Caffe Deep Learning Framework from Berkeley

init'd at a git revision because the current release is pretty old (~300 commits behind)

### CODE REVIEW

File: pkgs/applications/science/math/caffe/default.nix [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+, opencv
+, protobuf
+, snappy
+}:
+
+
+let optional = stdenv.lib.optional;
+in stdenv.mkDerivation rec {
+  # Use git revision because latest "release" is really old
+  name = "caffe-${rev}";
```
Comment (jagajaga) [inline pkgs/applications/science/math/caffe/default.nix]: Read this please https://nixos.org/nixpkgs/manual/#sec-package-naming and this https://github.com/NixOS/nixpkgs/pull/6539

Comment (jb55) [inline pkgs/applications/science/math/caffe/default.nix]: Thanks will fix

**Comment by jb55** [comment]
Name fixed :+1:

File: pkgs/applications/science/math/caffe/default.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}:
+
+
+let optional = stdenv.lib.optional;
+in stdenv.mkDerivation rec {
+  # Use git revision because latest "release" is really old
+  name = "caffe-git-2015-07-02";
+  rev = "77d66dfc";
+
+  src = fetchzip {
```
Comment (jagajaga) [inline pkgs/applications/science/math/caffe/default.nix]: Better use `fetchFromGithub`

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -14377,6 +14377,10 @@ let
 
   ### SCIENCE / MATH
 
+  caffe = callPackage ../applications/science/math/caffe {
+    cudaSupport = config.caffe.cudaSupport or config.cudaSupport or true;
```
Comment (jagajaga) [inline pkgs/top-level/all-packages.nix]: That's gonna be always `true`, it's ok?

Comment (jb55) [inline pkgs/top-level/all-packages.nix]: I would have thought so too, maybe I misunderstand how these expressions are evaluated but when I explicitly set config.cudaSupport to false it actually sets cudaSupport to false here.

File: pkgs/applications/science/math/caffe/default.nix (hunk 2 above)
Comment (jb55) [inline pkgs/applications/science/math/caffe/default.nix]: I've used that before but I was told to [use tarballs instead](https://github.com/NixOS/nixpkgs/pull/6907#discussion_r26823367), I'm guessing fetchFromGithub is preferred for revisions?

Comment (jagajaga) [inline pkgs/applications/science/math/caffe/default.nix]: fetchFromGithub does exactly the same, but it's interface is more human readable :)

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (jagajaga) [inline pkgs/top-level/all-packages.nix]: :open_mouth: 
Ok. Leave as it is.

**Comment by jb55** [comment]
Updated with `fetchFromGitHub` :+1:

**Pull request merged**